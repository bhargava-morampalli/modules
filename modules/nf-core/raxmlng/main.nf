process RAXMLNG {
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/raxml-ng:1.0.3--h32fcf60_0' :
        'biocontainers/raxml-ng:1.0.3--h32fcf60_0' }"

    input:
    path alignment

    output:
    path "*.raxml.bestTree", emit: phylogeny
    path "*.raxml.support" , optional:true, emit: phylogeny_bootstrapped
    path "versions.yml"    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def model = task.ext.model ?: 'GTR+G'
    """
    raxml-ng \\
        $args \\
        --msa $alignment \\
        --model $model \\
        --threads $task.cpus \\
        --prefix output

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        raxmlng: \$(echo \$(raxml-ng --version 2>&1) | sed 's/^.*RAxML-NG v. //; s/released.*\$//')
    END_VERSIONS
    """
    stub:
    def prefix = task.ext.prefix ?: "output"
    """
    touch ${prefix}.raxml.bestTree

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        raxmlng: \$(echo \$(raxml-ng --version 2>&1) | sed 's/^.*RAxML-NG v. //; s/released.*\$//')
    END_VERSIONS
    """
}
