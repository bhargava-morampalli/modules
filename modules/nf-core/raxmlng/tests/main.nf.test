nextflow_process {
  name "Test Process RAXMLNG"
  script "../main.nf"
  process "RAXMLNG"
  tag "modules"
  tag "modules_nfcore"
  tag "raxmlng"

  test("raxmlng - main") {
    when {
      process {
        """
        // alignment fasta file is required
        input[0] = [ // alignment fasta
          file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/alignment/informative_sites.fas', checkIfExists: true)
        ]
        """
      }

    }

    then {
      assertAll (
        { assert process.success },
        { assert process.out.phylogeny.size() > 0 },
        { assert process.out.phylogeny[0].toString().contains(".raxml.bestTree") },
        { assert process.out.versions.size() > 0 },
        { assert snapshot(process.out.versions).match("versions") }
      )
    }
  }

  test("raxmlng - stub") {
    options "-stub"
    when {
      process {
        """
        // alignment fasta file is required
        input[0] = [ // alignment fasta
          file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/alignment/informative_sites.fas', checkIfExists: true)
        ]
        """
      }
    }

    then {
      assertAll (
        { assert process.success },
        { assert snapshot(process.out).match() }
      )
    }
  }
}