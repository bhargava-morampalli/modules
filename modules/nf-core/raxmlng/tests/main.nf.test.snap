{
    "raxmlng - stub": {
        "content": [
            {
                "0": [
                    "output.raxml.bestTree:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,e5a855c4c9bcde2084ec9e34861379f0"
                ],
                "phylogeny": [
                    "output.raxml.bestTree:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "phylogeny_bootstrapped": [
                    
                ],
                "versions": [
                    "versions.yml:md5,e5a855c4c9bcde2084ec9e34861379f0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T15:15:19.072445918"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,e5a855c4c9bcde2084ec9e34861379f0"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T15:16:12.473313201"
    }
}